### PR TITLE
Allow user to override default for EMULATED_FUNCTION_POINTERS

### PR DESCRIPTION
**Pull request by sbc100**
If the user explicitly asks for EMULATED_FUNCTION_POINTERS=0 on the
command line we were not honoring that.

Partial workaround for #5370

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1151,13 +1151,15 @@ def check(input_file):
           # also tiny, and should be elimitated by meta-DCE when not used.
           shared.Settings.EXPORTED_FUNCTIONS += ['_setTempRet0', '_setThrew']
 
-      assert not (not shared.Settings.DYNAMIC_EXECUTION and shared.Settings.RELOCATABLE), 'cannot have both DYNAMIC_EXECUTION=0 and RELOCATABLE enabled at the same time, since RELOCATABLE needs to eval()'
+      if shared.Settings.RELOCATABLE and not shared.Settings.DYNAMIC_EXECUTION:
```
Comment (jgravelle-google) [inline emcc.py]: Not equivalent, https://en.wikipedia.org/wiki/De_Morgan%27s_laws
tl;dr, `!(A && B) == (!A) || (!B)`
so in this case `!(!A && B) == !!A || !B == A || !B`
so, `if shared.Settings.RELOCATABLE or not shared.Settings.DYNAMIC_EXECUTION`

Comment (jgravelle-google) [reply emcc.py]: For future historians, we talked in person and I was confused about the `!(!A && B)` -> `B && !A` swap. I also missed the conditional flip from the assert to the if. This is fine as-is.
